src/internal/helper.ts

```diff
@@ -0,0 +1,592 @@
+/*
+ * MinIO Javascript Library for Amazon S3 Compatible Cloud Storage, (C) 2015 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as crypto from 'node:crypto'
+import * as stream from 'node:stream'
+
+import { isBrowser } from 'browser-or-node'
+import { XMLParser } from 'fast-xml-parser'
+import ipaddr from 'ipaddr.js'
+import _ from 'lodash'
+import mime from 'mime-types'
+
+import type { Binary, Encryption, MetaData, ResponseHeader } from './type.ts'
+import { ENCRYPTION_TYPES } from './type.ts'
+
+/**
+ * All characters in string which are NOT unreserved should be percent encoded.
+ * Unreserved characters are : ALPHA / DIGIT / "-" / "." / "_" / "~"
+ * Reference https://tools.ietf.org/html/rfc3986#section-2.2
+ */
+export function uriEscape(string: string) {
```

Comment:
Should we explicitly declares the return type of the function?

Reply:
Unless @prakashsvmx also agrees that we can let the return types be inferred from usage, then so be it.

Reply:
TS can infer return type here. But I'm OK add return type too all functions/methods

Reply:
We can take it based on community feedback. for now it is fine as is.